src/compiler/parser.ts

```diff
@@ -4545,7 +4545,18 @@ namespace ts {
                         parseErrorAtRange(openingTag, Diagnostics.JSX_fragment_has_no_corresponding_closing_tag);
                     }
                     else {
-                        parseErrorAtRange(openingTag.tagName, Diagnostics.JSX_element_0_has_no_corresponding_closing_tag, getTextOfNodeFromSourceText(sourceText, openingTag.tagName));
+                        // We want the error to span only property identifier "Foo.Bar" e.g in < Foo.Bar >...
+                        const tag = openingTag.tagName as any;
+                        let start = tag.pos;
+                        if (tag.kind === SyntaxKind.PropertyAccessExpression){
+                            start = tag.expression.end - (tag.expression.escapedText.length);
+                        }
+                        // We want the error to span only tag identifier "Foo" e.g in < Foo >...
+                        else if(tag.kind === SyntaxKind.Identifier) {
+                            start = tag.end - tag.escapedText.length;
+                        }
+
+                        parseErrorAtRange({ pos: start, end: tag.end }, Diagnostics.JSX_element_0_has_no_corresponding_closing_tag, getTextOfNodeFromSourceText(sourceText, openingTag.tagName));

```

Comment:
We already have a [`parseErrorAt`](https://github.com/AGMETEOR/TypeScript/blob/e43276bce34507220a2e93f1713910753cbb6584/src/compiler/parser.ts#L1147) which takes a start/end pair.

Reply:
I have changed this as well to use parseErrorAt. Thanks for reviewing :)